test/test_linalg.py

```diff
@@ -921,6 +922,39 @@ def test_nuclear_norm_exceptions_old(self, device):
         self.assertRaisesRegex(RuntimeError, "duplicate or invalid", torch.norm, x, "nuc", (0, 0))
         self.assertRaisesRegex(IndexError, "Dimension out of range", torch.norm, x, "nuc", (0, 2))
 
+    @skipCUDAIfNoMagma
+    @skipCPUIfNoLapack
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
+    @dtypes(torch.float, torch.double, torch.cfloat, torch.cdouble)
+    @dtypesIfCPU(torch.float, torch.double, torch.cfloat, torch.cdouble)
+    @dtypesIfCUDA(torch.float, torch.double)
+    @precisionOverride({torch.float: 1e-4, torch.cfloat: 1e-4})
+    def test_tensorsolve(self, device, dtype):
+        def run_test(a_shape, axes):
+            a = torch.rand(a_shape, dtype=dtype, device=device)
+            b = torch.rand(a_shape[:2], dtype=dtype, device=device)
+            result = torch.linalg.tensorsolve(a, b, axes=axes)
+            expected = np.linalg.tensorsolve(a.cpu().numpy(), b.cpu().numpy(), axes=axes)
+            self.assertEqual(result, expected)
+
+        a_shapes = [(2, 3, 6), (3, 4, 4, 3)]
```

Comment:
Are there odd/extremal edge cases to test, too, like how empty tensors (on both the lhs and rhs) are handled?